### PR TITLE
Added support to the meta-intel-edison-bsp and meta-intel-edison-distro layers for the master(warrior) branch

### PR DESCRIPTION
**Pull request by nishantpoorswani**
Signed-off-by: nishantpoorswani <nishantpoorswani@gmail.com>

The master branch in Yocto has now moved from thud to warrior and hence, the image for the edison board wasn't compiling. I've added the warrior branch to the ` LAYERSERIES_COMPAT` variable in both the meta-intel-edison and meta-intel-edison-distro layers. This now compiles successfully with the master Yocto branch.

### CODE REVIEW

**Comment by htot** [comment]
Thanks. I am traveling now, but check this when I return.

**Comment by alext-mkrs** [comment]
While this is simple and looks good to me, based on the past experience, this won't be enough to migrate the layer to `warrior`. This setting is just to stop `bitbake` from complaining about layer incompatibility with the specific Yocto release, but it doesn't magically _make_ it compatible.

**Comment by nishantpoorswani** [comment]
Hi @alext-mkrs , I built an image for the Intel edison on the master branch using the **poky** distro and it works fine. For meta-intel-edison-distro layer, more changes will be required depending on the package bumps for warrior.

**Comment by nishantpoorswani** [comment]
A quick update, I've done with the other changes as well. Now edison-minimal builds with the poky-edison distro. I'll be sending the patches soon. Should I make a separate Pull request for those patches?

**Comment by htot** [comment]
There is an earlier PR from @PieterGIt here: https://github.com/htot/meta-intel-edison/pull/2/commits. It needs re-basing, but the first part is concerned about upgrading to warrior. The 2nd part is to build debian, which as I understand is a WIP. Maybe you can re-use in part?

**Comment by nishantpoorswani** [comment]
Hi @htot I've made the changes for the master branch. Sorry for the delay. I compiled the edison-image and it works perfectly. Could you please review it?

**Comment by htot** [comment]
@nishantpoorswani thanks I certainly will. 2 things are delaying me:
1 temperature in my attic where PC and Edison live is too high to play
2 extremely overloaded at work (expect to resolve after Friday)

**Comment by nishantpoorswani** [comment]
@htot whenever possible

**Comment by nishantpoorswani** [comment]
Hi @htot Any updates on this?

**Comment by htot** [comment]
I have been working on meta-acpi patches first and hope to get to this this weekend. Thanks for your patience.

**Comment by htot** [comment]
@nishantpoorswani applying your patches to master will upgrade that to warrior, so I want to release thud first. I have been working to get everything in place for that to happen this weekend, just need to test it before pushing to htot/meta-intel-edison. Then I'll send a PR to edison-fw for that. And create a warrior branch and pull your patches to htot/meta-intel-edison.

**Comment by nishantpoorswani** [comment]
@htot Sounds good to me

File: meta-intel-edison-distro/conf/distro/poky-edison.conf [hunk 1]
```diff
@@ -6,12 +6,12 @@ PREFERRED_VERSION_linux-yocto = "5.%"
 PREFERRED_PROVIDER_virtual/bootloader ?=  "u-boot"
 PREFERRED_VERSION_u-boot ?= "2019.04%"
 PREFERRED_VERSION_u-boot-fw-utils ?= "2019.04%"
-PREFERRED_VERSION_nodejs = "8.%"
-PREFERRED_VERSION_nodejs-native = "8.%"
+PREFERRED_VERSION_nodejs = "10.%"
+PREFERRED_VERSION_nodejs-native = "10.%"
 BINDINGS_pn-mraa="python nodejs"
 BINDINGS_pn-upm="python nodejs"
```
Comment (htot) [inline meta-intel-edison-distro/conf/distro/poky-edison.conf]: But on warrior (nodejs 10) upm enabling bindings cause an error due to IdleNotification deprecated and I don't know how to fix. @alext-mkrs ?

File: meta-intel-edison-bsp/recipes-bsp/u-boot/u-boot_2019.04.bb [hunk 2]
```diff
@@ -4,3 +4,16 @@ require u-boot-target-env.inc
...
+
+do_configure () {
+    if [ -z "${UBOOT_CONFIG}" ]; then
+        if [ -n "${UBOOT_MACHINE}" ]; then
+            oe_runmake -C ${S} O=${B} ${UBOOT_MACHINE}
+        else
+            oe_runmake -C ${S} O=${B} oldconfig
+        fi
+	echo ${PWD}
+        ${S}/scripts/kconfig/merge_config.sh -m .config ${@" ".join(find_cfgs(d))}
```
Comment (htot) [inline meta-intel-edison-bsp/recipes-bsp/u-boot/u-boot_2019.04.bb]: Send a patch with this line to ml?

Comment (nishantpoorswani) [reply meta-intel-edison-bsp/recipes-bsp/u-boot/u-boot_2019.04.bb]: I think we can definitely send it to oe-core ml. It's quite weird actually that nobody has complained or filed a bug about this.

Comment (htot) [reply meta-intel-edison-bsp/recipes-bsp/u-boot/u-boot_2019.04.bb]: Ok I'll do that.

**Comment by htot** [comment]
I pushed a new warrior branch https://github.com/htot/meta-intel-edison/tree/warrior with above patches and some more. @nishantpoorswani would you care to test this? It builds a 64bit kernel. I works for me, but I reordered patches after building and changed "u-boot: Fix path to merge_config.sh" to ed139db5 without testing.

Also I'd still like to hear from @alext-mkrs if we lose a lot by disabling upm nodejs bindings for now?

**Comment by nishantpoorswani** [comment]
@htot The patch for u-boot: Fix path to merge_config.sh doesn't apply because there's a slight mistake in the naming in the setup.sh(Line 348). The name should be 0001-u-boot-Fix-path-to-merge_config.sh.patch instead of 0001-u-boot-Fix-path-to-merge_config.sh

**Comment by htot** [comment]
@nishantpoorswani Sorry, that's a dumb mistake. Fixing now.

**Comment by nishantpoorswani** [comment]
@htot I've tested the https://github.com/htot/meta-intel-edison/tree/warrior branch. It builds perfectly.

**Comment by htot** [comment]
@nishantpoorswani thanks. I ran the image from usb too without problems. Now locally I have with u-boot 2019-7 and kernel 5.3. I find a small issue there with dwc3-pci that I hope to resolve. I hope in the mean time we can release thud and then pull warrior into master.

**Comment by htot** [comment]
@nishantpoorswani is reboot working for you? I updated to linux 5.3 and it is not, but I forgot to check proper when still on 5.2. I see your patch is applied on `b/units/systemd-reboot.service.in`, but on the image is only an unpatched `/lib/systemd/systemd-reboot.service`.

On 5.3 reboot hangs until watchdog reset. Manually copying in your patch to `/lib/systemd/systemd-reboot.service` reboot doesn't do anything at all.

Any ideas?

**Comment by alext-mkrs** [comment]
@htot, as far as UPM bindings are concerned - well, if it doesn't work, it doesn't work :) I don't see anything reported in the UPM repo, so probably upstream doesn't know yet. Probably worth reporting there.

**Comment by htot** [comment]
@alext-mkrs Upstream master switched to 2.0 :-)

**Comment by htot** [comment]
I pushed these patches with fixes and additional patches to https://github.com/htot/meta-intel-edison (master for 32 bits, warrior for 64 bits). So I will close this now. `htot/meta-intel-edison` will eventually be pulled into `edison-fw/meta-intel-edison` after review by @alext-mkrs.

Thanks @nishantpoorswani for your work.

**Comment by nishantpoorswani** [comment]
Hi @htot Sorry for the late reply. I've been really busy lately. I hadn't tested the reboot but I'm glad that you've managed to get it running. I'm really glad that I could help :)

**Pull request closed without merging**